wire-ios/Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchResultsViewController.swift

```diff
@@ -264,37 +254,30 @@ final class SearchResultsViewController: UIViewController {
         updateVisibleSections()
 
         searchResultsView.emptyResultContainer.isHidden = !isResultEmpty
-        searchDirectory?.updateIncompleteMetadataIfNeeded()
     }
 
-    @objc
-    func cancelPreviousSearch() {
+    private func performSearch(
+        query: String,
+        options: SearchOptions
+    ) {
         pendingSearchTask?.cancel()
         pendingSearchTask = nil
-    }
-
-    private func performSearch(query: String, options: SearchOptions) {
-        let selfUser = userSession.selfUser
-
-        pendingSearchTask?.cancel()
         searchResultsView.emptyResultContainer.isHidden = true
 
-        var options = options
-        options.updateForSelfUserTeamRole(selfUser: selfUser)
+        pendingSearchTask = Task {
```

Comment:
is there a specific reason we need to hold on to this task ? Looks like the purpose is to cancel it when we perform a new search but this cancellation is already handled in the `SearchUsersUseCase` so I wonder why we need to do it twice.

Reply:
as discussed, we probably need to hold on to this task since the parent task should be responsible of cancelling its child task, resolving my comment